Mods/ArchipelagoCore.SC2Mod/Base.SC2Data/Lib5BD4895D.galaxy

```diff
@@ -1367,6 +1517,151 @@ text lib5BD4895D_gf_AP_Core_objectiveText (text lp_text, int lp_checks, bool lp_
+
+    // Automatic Variable Declarations
+    int auto23B4A4BE_ae;
+    const int auto23B4A4BE_ai = 1;
+    int autoF9E8AC3D_ae;
+    const int autoF9E8AC3D_ai = 1;
+    int auto33CF2460_ae;
+    const int auto33CF2460_ai = 1;
+
+    // Variable Initialization
+
+    // Implementation
+    auto23B4A4BE_ae = (lib5BD4895D_gv_aP_Core_objectiveUsedCategories - 1);
+    lv_categoryindex = 0;
+    for ( ; ( (auto23B4A4BE_ai >= 0 && lv_categoryindex <= auto23B4A4BE_ae) || (auto23B4A4BE_ai < 0 && lv_categoryindex >= auto23B4A4BE_ae) ) ; lv_categoryindex += auto23B4A4BE_ai ) {
+        autoF9E8AC3D_ae = (lib5BD4895D_gv_aP_Core_objectivePanel.lv_categories[lv_categoryindex].lv_rollingIndex - 1);
+        lv_objectiveindex = 0;
+        for ( ; ( (autoF9E8AC3D_ai >= 0 && lv_objectiveindex <= autoF9E8AC3D_ae) || (autoF9E8AC3D_ai < 0 && lv_objectiveindex >= autoF9E8AC3D_ae) ) ; lv_objectiveindex += autoF9E8AC3D_ai ) {
+            auto33CF2460_ae = (lib5BD4895D_gv_aP_Core_LOCATIONS_PER_MAP - 1);
+            lv_locationindex = 0;
+            for ( ; ( (auto33CF2460_ai >= 0 && lv_locationindex <= auto33CF2460_ae) || (auto33CF2460_ai < 0 && lv_locationindex >= auto33CF2460_ae) ) ; lv_locationindex += auto33CF2460_ai ) {
+                if ((lib5BD4895D_gv_aP_Core_objectivePanel.lv_categories[lv_categoryindex].lv_objectives[lv_objectiveindex].lv_locations[lv_locationindex] == lp_locationID)) {
+                    return lib5BD4895D_gv_aP_Core_objectivePanel.lv_categories[lv_categoryindex].lv_objectives[lv_objectiveindex].lv_objective;
+                }
+
+            }
+        }
+    }
+    return c_invalidObjectiveId;
+}
+
+void lib5BD4895D_gf_AP_Core_associateObjectiveWithLocation (int lp_objective, int lp_locationID) {
+    // Variable Declarations
+    int lv_categoryindex;
+    int lv_objectiveindex;
+    int lv_locationindex;
+
+    // Automatic Variable Declarations
+    int autoFF749E50_ae;
+    const int autoFF749E50_ai = 1;
+
+    // Variable Initialization
+    lv_categoryindex = lib5BD4895D_gf_AP_Core_objectiveCategoryIndex(lp_objective);
+    lv_objectiveindex = lib5BD4895D_gf_AP_Core_objectiveIndex(lp_objective, lv_categoryindex);
+
+    // Implementation
+    if (((lp_locationID <= -1) || (lib5BD4895D_gv_aP_Core_locationCollectable[lp_locationID] == false))) {
+        return ;
+    }
+
+    autoFF749E50_ae = (lib5BD4895D_gv_aP_Core_LOCATIONS_PER_MAP - 1);
+    lv_locationindex = 0;
+    for ( ; ( (autoFF749E50_ai >= 0 && lv_locationindex <= autoFF749E50_ae) || (autoFF749E50_ai < 0 && lv_locationindex >= autoFF749E50_ae) ) ; lv_locationindex += autoFF749E50_ai ) {
+        if ((lib5BD4895D_gv_aP_Core_objectivePanel.lv_categories[lv_categoryindex].lv_objectives[lv_objectiveindex].lv_locations[lv_locationindex] == -1)) {
+            lib5BD4895D_gv_aP_Core_objectivePanel.lv_categories[lv_categoryindex].lv_objectives[lv_objectiveindex].lv_locations[lv_locationindex] = lp_locationID;
+            return ;
+        }
+
+    }

```

Comment:
May be worth sticking some kind of debug / printout here. Unlikely to matter, but in the case it does it may be helpful for a dev to track down an issue.

I'm not sure what a good channel of communication is. Could be a chat message, tip, or AP message box message, though they all have downsides. May be worth asking @Ziktofel or @Salzkorn if we have any existing debug logging stuff somewhere.

Reply:
Good idea. I've added an error notification (appears on left-side of screen) stating: "Error: Failed to associate Location ID # to objective". Given this, one can easily search up the ID in the triggers and figure out what's going on.